src/NServiceBus.Core/Transports/Msmq/MsmqUtilities.cs

```diff
@@ -71,6 +71,34 @@ static MsmqAddress GetIndependentAddressForQueue(MessageQueue q)
             return headers;
         }
 
+        public static bool TryOpenQueue(string address, out MessageQueue messageQueue)
+        {
+            messageQueue = null;
+            var msmqAddress = MsmqAddress.Parse(address);
+            var path = msmqAddress.PathWithoutPrefix;
+            try
+            {
+                if (MessageQueue.Exists(path))
+                {
+                    messageQueue = new MessageQueue(path);
+                    return true;
+                }
+            }
+            catch (MessageQueueException mex)
+            {
+                if (msmqAddress.IsRemote && (mex.MessageQueueErrorCode == MessageQueueErrorCode.IllegalQueuePathName))
+                {
+                    return false;
+                }
+                if (mex.MessageQueueErrorCode == MessageQueueErrorCode.QueueExists)
+                {
+                    return false;
+                }
```

Comment:
@bording you're right of course, I'll change that. But I'm also making this PR wip for now, I'm not happy with this in general. I'll adjust it.